Modules/_decimal/_decimal.c

```diff
@@ -3676,20 +3676,20 @@ dec_as_long(PyObject *dec, PyObject *context, int round)
     }
 
     n = (mpd_sizeinbase(x, 2) + PyLong_SHIFT - 1)/PyLong_SHIFT;
-    PyLongWriter *writer = PyLongWriter_Create(mpd_isnegative(x), n,
-                                               (void**)&ob_digit);
-    /* mpd_sizeinbase can overestimate size by 1 digit, set it to zero. */
-    ob_digit[n-1] = 0;
+    PyLongWriter *writer = PyLongWriter_Create(mpd_isnegative(x), n, &digits);
     if (writer == NULL) {
         mpd_del(x);
         return NULL;
     }
 
     status = 0;
+    /* mpd_sizeinbase can overestimate size by 1 digit, set it to zero. */
 #if PYLONG_BITS_IN_DIGIT == 30
```

Comment:
Would it make sense to use `uint32_t *uint32_digits = (uint32_t *)digits;` and `uint32_digit[n-1] = 0;`?

Reply:
You can then use `uint32_digits` in `mpd_qexport_u32()`. It may make to code cleaner.